DirectProgramming/DPC++FPGA/PlatformTests/board_test/README.md

```diff
@@ -0,0 +1,341 @@
+
+Above interface checks are divided into six sub-tests. By default, all tests are run. You can choose to run a single test using `-test=<test number>` option. Please see [Running the Sample](#running-the-sample) for test usage instructions. 
+1. Host Speed and Host Read Write Test
+2. Kernel Clock Frequency Test
+3. Kernel Launch Test
+4. Kernel Latency Measurement
+5. Kernel-to-Memory Read Write Test
+6. Kernel-to-Memory Bandwidth Test<br>
+*Note:* It is recommended to run all tests at least once to ensure that the platform interfaces are fully functional.
+
+## Key Concepts
+* Sample to demonstrate platform interface checks
+
+## License  
+Code samples are licensed under the MIT license. See
+[License.txt](https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt) for details.
+
+Third party program Licenses can be found here: [third-party-programs.txt](https://github.com/oneapi-src/oneAPI-samples/blob/master/third-party-programs.txt)
+
+## Building the `board_test` code sample
+
+### Include Files
+The included header `dpc_common.hpp` is located at `$ONEAPI_ROOT/dev-utilities/latest/include` on your development system.
+
+### Running Samples in DevCloud
+If running a sample in the Intel DevCloud, remember that you must specify the type of compute node and whether to run in batch or interactive mode. Compiles to FPGA are only supported on fpga_compile nodes. Executing programs on FPGA hardware is only supported on fpga_runtime nodes of the appropriate type, such as fpga_runtime:arria10 or fpga_runtime:stratix10.  Neither compiling nor executing programs on FPGA hardware are supported on the login nodes. For more information, see the [Intel® oneAPI Base Toolkit Get Started Guide](https://devcloud.intel.com/oneapi/documentation/base-toolkit/).
+
+When compiling for FPGA hardware, it is recommended to increase the job timeout to 12h.
+
+### On a Linux* System
+
+1. Generate the `Makefile` by running `cmake`.
+     ```
+   mkdir build
+   cd build
+   ```
+   To compile for the Intel® FPGA PAC D5005 (with Intel Stratix® 10 SX), run `cmake` using the command:
+    ```
+    cmake ..
+   ```
+   Alternatively, to compile for the Intel® PAC with Intel Arria® 10 GX FPGA, run `cmake` using the command:
+
+   ```
+   cmake .. -DFPGA_BOARD=intel_a10gx_pac:pac_a10
+   ```
+   You can also compile for a custom FPGA platform. Ensure that the board support package is installed on your system. Then run `cmake` using the command:
+   ```
+   cmake .. -DFPGA_BOARD=<board-support-package>:<board-variant>
+   ```
+
+2. Compile the design through the generated `Makefile`. The following build targets are provided, matching the recommended development flow:
+
+   * Compile for emulation (compiles quickly, targets emulated FPGA device):
+      ```
+      make fpga_emu
```

Comment:
I tried this in ARC using 2021.3 and 2021.4, and I got this error:

```
board_test $> mkdir build
board_test $> cd build/
build $> cmake ..
-- The CXX compiler identification is Clang 13.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /p/psg/swip/hld/r/sycl/rel/20210619/linux64/bin/dpcpp - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- FPGA_BOARD was not specified.                    
Configuring the design to run on the default FPGA board intel_s10sx_pac:pac_s10 (Intel(R) PAC with Intel Stratix(R) 10 SX FPGA).                     
Please refer to the README for information on board selection.
-- Configuring done
-- Generating done
-- Build files have been written to: /p/psg/swip/w/whitepau/oneAPI_Experiments/anuja-oneAPI-samples/DirectProgramming/DPC++FPGA/PlatformTests/board_test/build
build $> make fpga_emu
Scanning dependencies of target board_test.fpga_emu
[ 50%] Building CXX object src/CMakeFiles/board_test.fpga_emu.dir/board_test.cpp.o
/p/psg/swip/w/whitepau/oneAPI_Experiments/anuja-oneAPI-samples/DirectProgramming/DPC++FPGA/PlatformTests/board_test/src/board_test.cpp:31:10: fatal error: 'dpc_common.hpp' file not found
#include "dpc_common.hpp"
         ^~~~~~~~~~~~~~~~
1 error generated.
src/CMakeFiles/board_test.fpga_emu.dir/build.make:81: recipe for target 'src/CMakeFiles/board_test.fpga_emu.dir/board_test.cpp.o' failed
make[3]: *** [src/CMakeFiles/board_test.fpga_emu.dir/board_test.cpp.o] Error 1
CMakeFiles/Makefile2:253: recipe for target 'src/CMakeFiles/board_test.fpga_emu.dir/all' failed
make[2]: *** [src/CMakeFiles/board_test.fpga_emu.dir/all] Error 2
CMakeFiles/Makefile2:206: recipe for target 'src/CMakeFiles/fpga_emu.dir/rule' failed
make[1]: *** [src/CMakeFiles/fpga_emu.dir/rule] Error 2
Makefile:176: recipe for target 'fpga_emu' failed
make: *** [fpga_emu] Error 2
```

Reply:
Cleaner compilers now, warnings resolved.